src/Utils/Image.php

```diff
@@ -267,6 +271,8 @@ public static function detectTypeFromString(string $s, &$width = null, &$height
 
 	/**
 	 * Returns the file extension for the given `Image::XXX` constant.
+	 * @param self::JPEG|self::PNG|self::GIF|self::WEBP|self::AVIF|self::BMP $type
+	 * @return value-of<self::Formats>
```

Comment:
What's `@return` annotation good for?

Reply:
For me the most important thing out of this is specifying that the output extension is non-empty string, which is useful when you want to use the extension further. I guess `@return non-empty-string` could cover most of the use-cases, but why shouldn't we make the return type as narrow as possible?

Reply:
Now, as a return annotation, this should probably be `ImageType::*`, right?

Reply:
No, because it returns one of the string values contained in `self::Formats`, not `ImageType::*` value.